pages/[slug].tsx

```diff
@@ -0,0 +1,45 @@
+import type { NextPage } from 'next';
+import type { AreWeHeadlessYetTopicPage } from '../components/types';
+import type { Topic } from '../components/StreamField/blocks/TopicsBlock';
+
+import {
+    getAreWeHeadlessYetTopicPages,
+    getAreWeHeadlessYetTopicPage,
+} from '../lib';
+
+import Layout from '../components/Layout';
+import TopicPageHero from '../components/TopicPageHero/TopicPageHero';
+import StreamField from '../components/StreamField';
+
+const TopicPage: NextPage<{ page: AreWeHeadlessYetTopicPage }> = ({ page }) => {
+    return (
+        <Layout title={page.title} lastPublishedAt={page.lastPublishedAt}>
+            <TopicPageHero {...page} />
+            <StreamField body={page.body} />
+        </Layout>
+    );
+};
+
+export async function getStaticPaths() {
+    const topics = await getAreWeHeadlessYetTopicPages();
+    const paths = topics.map((topic: Topic) => ({
+        params: {
+            slug: topic.meta.slug,
+        },
+    }));
+
+    return {
+        paths,
+        fallback: false,
+    };
+}
+
+export async function getStaticProps({ params }: { [key: string]: any }) {
+    const page = await getAreWeHeadlessYetTopicPage(params.slug);
+
+    return {
+        props: { page: page },
+    };
+}
+
+export default TopicPage;
```

Comment:
**Question**

I'm not familiar with NextJS - is this file a NextJS-specific thing? 

Also what're the functions `getStaticPaths` and `getStaticProps` for? I don't see them used anywhere in this PR.

Reply:
This file is what's rendered for each topic - See [dynamic routing](https://nextjs.org/docs/routing/dynamic-routes).

Those functions are used for [static generation with data](https://nextjs.org/docs/basic-features/pages#static-generation-with-data).